vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDeviceProvider.kt

```diff
@@ -167,7 +167,7 @@ class AndroidDeviceProvider : DeviceProvider, CoroutineScope {
     private fun matchDdmsToDevice(device: IDevice): AndroidDevice? {
         val observedDevices = devices.values
         return observedDevices.findLast {
```

Comment:
with `devices` being a `ConcurrentHashMap` instance,  shouldn't this use some concurrent retrieval method instead?

Reply:
Actually not, at that point we are looking if there is just the same device in a Map - no need to block it.